lib/pact/command/cont_command.ex

```diff
@@ -0,0 +1,333 @@
+    do: %{cmd_request | nonce: nonce}
+
+  def set_nonce(%__MODULE__{}, _nonce), do: {:error, [nonce: :not_a_string]}
+  def set_nonce({:error, reason}, _nonce), do: {:error, reason}
+
+  @impl true
+  def set_metadata(%__MODULE__{} = cmd_request, %MetaData{} = meta_data),
+    do: %{cmd_request | meta_data: meta_data}
+
+  def set_metadata(%__MODULE__{} = cmd_request, meta_data) do
+    case MetaData.new(meta_data) do
+      %MetaData{} = meta_data -> %{cmd_request | meta_data: meta_data}
+      {:error, reason} -> {:error, [meta_data: :invalid] ++ reason}
+    end
+  end
+
+  def set_metadata({:error, reason}, _metadata), do: {:error, reason}
+
+  @impl true
+  def add_keypair(%__MODULE__{keypairs: keypairs} = cmd_request, %KeyPair{} = keypair),
+    do: %{cmd_request | keypairs: keypairs ++ [keypair]}
+
+  def add_keypair(%__MODULE__{} = cmd_request, keypair) do
+    case KeyPair.new(keypair) do
+      %KeyPair{} = keypair -> add_keypair(cmd_request, keypair)
+      {:error, reason} -> {:error, [keypair: :invalid] ++ reason}
+    end
+  end
+
+  def add_keypair({:error, reason}, _keypair), do: {:error, reason}
+  @impl true
+  def add_keypairs(%__MODULE__{} = cmd_request, []), do: cmd_request
+
+  def add_keypairs(%__MODULE__{} = cmd_request, [keypair | keypairs]) do
+    cmd_request
+    |> add_keypair(keypair)
+    |> add_keypairs(keypairs)
+  end
+
+  def add_keypairs(%__MODULE__{}, _keypairs), do: {:error, [keypairs: :not_a_list]}
+  def add_keypairs({:error, reason}, _keypairs), do: {:error, reason}
+
+  @impl true
+  def add_signer(%__MODULE__{signers: nil} = cmd_request, %Signer{} = signer),
+    do: %{cmd_request | signers: SignersList.new([signer])}
+
+  def add_signer(%__MODULE__{signers: signer_list} = cmd_request, %Signer{} = signer) do
+    %SignersList{signers: signers} = signer_list
+    %{cmd_request | signers: SignersList.new(signers ++ [signer])}
+  end
+
+  def add_signer(%__MODULE__{}, _signer), do: {:error, [signer: :invalid]}
+  def add_signer({:error, reason}, _signer), do: {:error, reason}
+
+  @impl true
+  def add_signers(%__MODULE__{} = cmd_request, %SignersList{} = list),
+    do: %{cmd_request | signers: list}
+
+  def add_signers(%__MODULE__{} = cmd_request, signers) do
+    case SignersList.new(signers) do
+      %SignersList{} = signers -> %{cmd_request | signers: signers}
+      {:error, reason} -> {:error, [signers: :invalid] ++ reason}
+    end
+  end
+
+  def add_signers({:error, reason}, _signers), do: {:error, reason}
+
+  @impl true
+  def set_pact_tx_hash(%__MODULE__{} = cmd_request, pact_tx_hash) when is_binary(pact_tx_hash) do
+    %{cmd_request | pact_tx_hash: pact_tx_hash}
+  end
+
+  def set_pact_tx_hash(%__MODULE__{}, _pact_tx_hash), do: {:error, [pact_tx_hash: :not_a_string]}
+  def set_pact_tx_hash({:error, reason}, _pact_tx_hash), do: {:error, reason}
+
+  @impl true
+  def set_step(%__MODULE__{} = cmd_request, step) when is_integer(step) do
+    %{cmd_request | step: step}
+  end
```

Comment:
Since the body only has one line, this block could be avoided and instead use the `, do:` notation__